backend/vehicles/src/modules/company-user-management/users/profiles/user.profile.ts

```diff
@@ -231,15 +231,33 @@ export class UsersProfile extends AutomapperProfile {
         forMember(
           (d) => d.userAuthGroup,
           mapFrom((s) => {
-            if (s.companyUsers?.length && s.companyUsers[0]?.userAuthGroup) {
-              //the logic to be revisited if the application decide to support
-              //one user id multiple companies
-              return s.companyUsers[0]?.userAuthGroup;
-            } else {
-              return s.userAuthGroup;
+            if (
+              s.companyUsers?.length > 0 &&
+              s.companyUsers[0]?.userAuthGroup
+            ) {
+              const companyUser = s.companyUsers.filter((i) => {
+                return i.userGUID === s.userGUID;
+              });
+              if (companyUser?.length) {
```

Comment:
a ternary might work here too:
```
return companyUser?.length ? companyUser[0].userAuthGroup : s.userAuthGroup;
```

Reply:
Might end up reworking the whole thing. we are thinking of fine tuning the DB query to return one company user. hence draft pull request.